### PR TITLE
removed multipart form response logic - using requests instead

### PR DESCRIPTION
**Pull request by lucagiovagnoli**
Fixes issue #199. 
I substituted logic that builds the request for FidoClient with python `requests` and refactored tests.

Unfortunately the tests here could not detect the 'content-length' bug (see comments). I found it by acceptance-testing the package with our framework. Opened PR26 on Yelp/fido to fix it upstream.

### CODE REVIEW

File: bravado/fido_client.py [hunk 1]
```diff
@@ -1,19 +1,15 @@
...
+import requests
 import six
 
 if six.PY3:
     raise ImportError("The fido client is not yet supported in py3")
 
 import fido
-from bravado_core.param import stringify_body as param_stringify_body
 from bravado_core.response import IncomingResponse
-from yelp_uri import urllib_utf8
```
Comment (laucia) [inline bravado/fido_client.py]: this is the only place where yelp_uri is used, you can remove the requirements from setup.py

File: bravado/fido_client.py [hunk 2]
```diff
@@ -1,19 +1,15 @@
 # -*- coding: utf-8 -*-
 import logging
-
+import requests
```
Comment (analogue) [inline bravado/fido_client.py]: Please retain the blank line between standard library imports and 3rd party imports.

File: bravado/fido_client.py [hunk 3]
```diff
@@ -1,19 +1,15 @@
...
 import six
 
 if six.PY3:
     raise ImportError("The fido client is not yet supported in py3")
 
 import fido
-from bravado_core.param import stringify_body as param_stringify_body
```
Comment (analogue) [inline bravado/fido_client.py]: Can you verify nothing else in bravado or bravado-core uses `bravado_core.param.stringify_body`? If not, we can ticket for removal.

File: bravado/fido_client.py [hunk 4]
```diff
@@ -70,36 +66,44 @@ def request(self, request_params, operation=None, response_callbacks=None,
...
+        concurrent_future = fido.fetch(**request_for_crochet)
 
         return HttpFuture(concurrent_future,
                           FidoResponseAdapter,
                           operation,
                           response_callbacks,
                           also_return_response)
 
+    @staticmethod
+    def prepare_request_for_crochet(request_params):
```
Comment (analogue) [inline bravado/fido_client.py]: Please document args and return type.

File: tests/fido_client/FidoClient/request_test.py [hunk 5]
```diff
@@ -1,53 +0,0 @@
...
-
-
-try:
-    from bravado.fido_client import FidoClient
-except ImportError:
-    FidoClient = Mock()
-
-
-@pytest.mark.skipif(six.PY3, reason="twisted doesnt support py3 yet")
-def test_no_timeouts_passed_to_fido():
```
Comment (analogue) [inline tests/fido_client/FidoClient/request_test.py]: Why all the moving of tests? The existing pattern in the lib is one file per method under test. Would like to keep this consistent.

File: bravado/fido_client.py (hunk 3 above)
Comment (lucagiovagnoli) [inline bravado/fido_client.py]: It seems that it is not used anywhere else:
https://github.com/Yelp/bravado/search?utf8=%E2%9C%93&q=stringify_body
https://github.com/Yelp/bravado-core/search?utf8=%E2%9C%93&q=stringify_body

Ticketed here https://github.com/Yelp/bravado-core/issues/91

**Comment by analogue** [comment]
:+1:

**Pull request merged**